elastic/kibana
# [Security Solution] updates generateId function to use timestamp

## Summary

updates generateId function to use ~~_seq_no and _primary_term~~ timestamp (instead of _version, which wasn't previously being properly received) to ensure that alerts are generated when a document is updated.

## Testing

1. create a test index
```
PUT test-index/_mapping
{
  "properties": {
    "@timestamp": {
      "type": "date"
    }
  }
}
```

2. add an event to the test index
```
POST test-index/_doc
{
  "@timestamp": 1637346550203,
  "test-id": 1
}
```
3. Create a custom query rule with `test-id: *` query and activate the rule to see an alert

4. Update the previously created event while the rule is running (timestamp must be changed)
```
PUT test-index/_doc/1
{
  "@timestamp": 1637346559999,
  "test-id": 1
}
```

5. confirm that there is a new alert generated after the update.

### Checklist

Delete any items that are not applicable to this PR.
